src/sinks/apex/mod.rs

```diff
@@ -0,0 +1,170 @@
+use bytes::Bytes;
+use futures::FutureExt;
+use futures_util::SinkExt;
+use http::{Request, StatusCode, Uri};
+use hyper::Body;
+use serde_json::json;
+use vector_config::configurable_component;
+
+#[cfg(feature = "apex-integration-tests")]
+#[cfg(test)]
+mod integration_tests;
+
+use crate::{
+    codecs::Transformer,
+    config::{
+        AcknowledgementsConfig, GenerateConfig, Input, SinkConfig, SinkContext, SinkDescription,
+    },
+    event::Event,
+    http::HttpClient,
+    sinks::util::{
+        http::{BatchedHttpSink, HttpEventEncoder, HttpSink},
+        BatchConfig, BoxedRawValue, JsonArrayBuffer, RealtimeSizeBasedDefaultBatchSettings,
+        TowerRequestConfig, UriSerde,
+    },
+};
+
+/// Configuration for the `apex` sink.
+#[configurable_component(sink)]
+#[derive(Clone, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct ApexSinkConfig {
+    /// The base URI of the Apex instance. Vector will append `/add_events` to this.
```

Comment:
```suggestion
    /// The base URI of the Apex instance.
    ///
    /// Vector will append `/add_events` to this.
```

This will end up rendering better, having a title and separate description.